README.md

```diff
@@ -1 +1,128 @@
+`['error' => 'error provided by Kafka', message => (Message object given to produce)]`.
+Whether the error is [non-]retryable is based on the `error_code` as documented
+[here](https://docs.confluent.io/current/kafka-rest/api.html#post--topics-(string-topic_name).
+It's up to you what you do with those.
+
+```php
+$producer = new Producer($restClient);
+$producer->produce('your-topic', new Message('some-message'));
+
+$messages = [new Message('some-message'), new Message('and-some-other-message')];
+$producer->produceBatch('your-topic', $messages);
+```
+
+### Consumer
+
+Consumer allows you to consume messages one by one until you return from the loop, application throws exception
+or is otherwise forced to exit.
+The `consume` method returns a [Generator](https://www.php.net/manual/en/class.generator.php) and loops indefinitely,
+yielding messages as they are available. `consume` method accepts two parameters: `timeout` and `maxBytes`.
+`timeout` is the maximum time the consumer will wait for the messages. `maxBytes` is the maximum size of the messages
+to fetch in a single request. Both of these settings are complementary to the settings in `ConsumerOptions` and to the
+server settings (check the Kafka documentation for more information).
+
+If you didn't set the consumer to auto-commit messages then you should use consumers `commit` method to do so. Simply
+pass it a message you wish to commit. For most cases it's recommended to leave auto-commit off and manually commit
+each message as you process them.
+
+```php
+$consumerFactory = new ConsumerFactory($restClient);
+$consumer = $consumerFactory->create('your-consumer-group', Subscription::topic('your-topic'));
+foreach ($consumer->consume() as $message) {
+    // Do your magic
+    $logger->info('got new message', $message->content);
+
+    // ... and when you are done, commit the message.
+    $consumer->commit($message);
+}
+```
+
+### BatchConsumer
+
+BatchConsumer works the same way as Consumer does; the difference is that BatchConsumer doesn't yield each message
+separately but first puts them in batches (`MessagesBatch`). These batches can be configured to be "limited" by either
+count of messages in the batch (`maxCount`), by time (`maxDuration`) or by both (setting both `maxCount` and
+`maxDuration`). If you set `maxDuration` then the batch won't ever take longer than that (+ few ms for processing) as
+it changes the `timeout` parameter of the consumer (consumer won't get stuck on network waiting for more messages).
+
+BatchConsumer can be quite useful for processing of "large" data sets, where you would have to otherwise batch
+the messages yourself (eg. for inserting into database, where batch operations are always better).
+
+As mention in the Consumer example, you may need to commit the messages. For that there is `commit` method,
+which accepts the yielded `MessagesBatch` and commits all the messages inside it in one request.
```

Comment:
commits last message's offset

Reply:
it will actually commit multiple offsets, I  think it's better to not go that deep and just stay with "your messages will be committed"

Reply:
Is it rest proxy specific thing? but sure

Reply:
no, you can consume multiple topics and multiple partitions ... you need to commit last offset of each partition-topic combo